programs/bubblegum/program/src/processor/update_metadata.rs

```diff
@@ -207,13 +169,14 @@ pub fn update_metadata<'info>(
     root: [u8; 32],
     nonce: u64,
     index: u32,
-    current_metadata: Option<MetadataArgs>,
+    current_metadata: MetadataArgs,
     update_args: UpdateArgs,
 ) -> Result<()> {
-    assert_signed_by_tree_delegate(&ctx.accounts.tree_authority, &ctx.accounts.tree_delegate)?;
-
-    // Determine how the user opted to pass in the old MetadataArgs
-    let current_metadata = fetch_current_metadata(current_metadata, &ctx.accounts.metadata_buffer)?;
+    require!(
+        ctx.accounts.tree_delegate.key() == ctx.accounts.tree_authority.tree_creator
+            || ctx.accounts.tree_delegate.key() == ctx.accounts.tree_authority.tree_delegate,
+        BubblegumError::TreeAuthorityIncorrect,
+    );
```

Comment:
Beyond removing the unneeded `fetch_current_metadata` function, I also just simplified out the `assert_signed_by_tree_delegate` function because it just seemed extra now that its only called once in this instruction.